pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -808,3 +810,38 @@ func getSNIHostsForServer(server *networking.Server) []string {
 
 	return sniHostsSlice
 }
+
+func buildGatewayVirtualHostDomains(hostname string, ports []int) []string {
+	domains := []string{hostname}
+	if hostname == "*" {
+		return domains
+	}
+
+	for _, port := range ports {
+		domains = append(domains, fmt.Sprintf("%s:%d", hostname, port))
+	}
+
+	return domains
+}
+
+func findServicePorts(node *model.Proxy, svcPort int) []int {
+	ports := []int{svcPort}
+	if len(node.ServiceInstances) == 0 {
+		return ports
+	}
+	svcPortName := ""
+	for _, p := range node.ServiceInstances[0].Service.Ports {
+		// found the svc port
+		if p.Port == svcPort {
+			svcPortName = p.Name
+			continue
+		}
+		// found another port, nodeport for k8s service
+		if svcPortName != "" && svcPortName == p.Name {
```

Comment:
this can never be reached if p.Port == svcPort

Reply:
Yes, this is just for nodeport.